duckduckgo/iOS
# Fire education iteration

Task/Issue URL: https://app.asana.com/0/72649045549333/1179359141574181/f
Tech Design URL:
CC: @cmonfortep 

**Description**:
iOS implementation of Android Fire Button education.
We originally tried an iteration of the android one, but the result wasn't statistically significant. So this instead implements the same flow as on Android (which saw a statistically significant retention lift).

This PR also removes the previous experiment and some logic that is no longer needed in FullScreenDaxDialogViewController

Now instead of showing the fire education dialog directly, it makes the fire button pulse (for one hour, or until it's tapped on or a tab is opened).
If the user presses the fire button whilst it's pulsing, then the fire education dialog is shown.
The logic is quite complicated, so LMK if you have any questions.

Things to note: 
- Pulse should start showing at generally the same place in the flow that the previous fire education started showing
- It should time out after an hour
- It should stop showing if the user opens a new tab
- Other actions (such as opening the settings menu) shouldn't cancel the pulse
- if it's within the hour window and you close the app and open it again, the pulse should show.
- The top and bottom bars shouldn't hide when scrolling if the pulse is visible, but should behaviour normally when it's not visible.
- If you press the fire button on the tab switcher screen, the dialog should show if the pulse is active.  Android does show the dialog here, but not the pulse because it was more complicated. On iOS it was easier to make it pulse here too and we decided we'd prefer that.

**Steps to test this PR**:
1. Test experiment is bucketed correctly (I.e. test that you see the fire education roughly 50% of the time)
2. Test nothing has changed when the experiment isn't active
3. Test favourites widget onboarding still works correctly
4. Test fire education dialog shows at the correct time. The pulse should start showing at the same time as the old fire dialog showed.
5. Test that hiding dax tips before seeing the fire button education works (i.e. fire button education shouldn't be shown)
6. Test pixels are fired when expected
7. Test the things in "Things to note" above



**Copy Testing**:


**Orientation Testing**:


**Device Testing**:


**OS Testing**:


**Theme Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
